### PR TITLE
Expose setAllowUniversalAccessFromFileURLs in Android WebView

### PR DESCRIPTION
**Pull request by locly**
This pull request exposes the `setAllowUniversalAccessFromFileURLs` method of Android WebViewSettings as a property. The reason for this is when loading pages with a `file://` baseUrl it's sometimes desirable to allow loading other assets from a file base url. (For example loading an image into a canvas). More information on its use and purpose can be found [in the android docs here](https://developer.android.com/reference/android/webkit/WebSettings.html#setAllowUniversalAccessFromFileURLs%28boolean%29)

Usage example:

``` jsx
return (
  <WebView
    source={{ html: myhtml, baseUrl: 'file://' }}
    allowUniversalAccessFromFileURLs={true}
    javaScriptEnabled={true} />
)
```

### CODE REVIEW

**Comment by facebook-github-bot-4** [comment]
By analyzing the blame information on this pull request, we identified @caabernathy and @nicklockwood to be potential reviewers.

**Comment by facebook-github-bot-4** [comment]
It's been a while since the last commit was reviewed and the labels show this pull request needs review. Based on the blame information for the files in this pull request we identified @spicyj as a potential reviewer. Could you take a look please or cc someone with more context?

**Comment by facebook-github-bot-2** [comment]
@locly updated the pull request - [view changes](https://github.com/facebook/react-native/pull/8905/files/4c8ee213fbe935b39744b5b5487f4865d50fbc76..f5192b75054389f0c28380f78099cf4af34bc3e9)

**Comment by thurber** [comment]
Just want to comment that I would love to see this PR move forward; I'm building an Android app that depends on this feature for displaying arbitrary PDFs loaded from the web, by embedding a copy of Mozilla's PDF.js in the Android assets.

File: Libraries/Components/WebView/WebView.ios.js [hunk 1]
```diff
@@ -324,6 +324,14 @@ class WebView extends React.Component {
...
      */
     mediaPlaybackRequiresUserAction: PropTypes.bool,
+
+    /**
+     * Boolean that sets whether JavaScript running in the context of a file
+     * scheme URL should be allowed to access content from any origin.
+     * Including accessing content from other file scheme URLs
+     * @platform android
+     */
+    allowUniversalAccessFromFileURLs: PropTypes.bool,
```
Comment (chirag04) [inline Libraries/Components/WebView/WebView.ios.js]: do don't need this here i guess.

**Comment by chirag04** [comment]
LGTM. cc @satya164

File: Libraries/Components/WebView/WebView.ios.js (hunk 1 above)
Comment (satya164) [inline Libraries/Components/WebView/WebView.ios.js]: yeah, remove it from here

File: Libraries/Components/WebView/WebView.android.js [hunk 2]
```diff
@@ -153,6 +153,13 @@ class WebView extends React.Component {
      * start playing. The default value is `false`.
      */
     mediaPlaybackRequiresUserAction: PropTypes.bool,
+
+    /**
+     * Boolean that sets whether JavaScript running in the context of a file
+     * scheme URL should be allowed to access content from any origin.
+     * Including accessing content from other file scheme URLs
+     */
+    allowUniversalAccessFromFileURLs: PropTypes.bool,
```
Comment (satya164) [inline Libraries/Components/WebView/WebView.android.js]: add `@platform android`

**Comment by hramos** [comment]
@locly still planning to ship this?

**Comment by jacobp100** [comment]
Sorry, I missed the notification for this. Any chance we could see how this would look on iOS? I’m concerned that people may come to depend on this API, then later we want to add iOS support, and we have to cause breaking changes.

**Comment by locly** [comment]
Hi, sorry we missed the notification on these. Definitely want it merged in though. Have re-synced with Master.

@satya164 @chirag04 have removed the prop from the ios version and added the `@android` comment.

@jacobp100 as regards to iOS, it's much more lax than Android and doesn't have any restrictions on file urls and no `UIWebView` properties to change as regards this

**Comment by jacobp100** [comment]
Alright. Should it be on by default then to make the platform’s default behaviour match? Your call on that. Other than that, LGTM.

**Comment by locly** [comment]
@jacobp100 on a platform agnostic level I would agree, but in terms of Android this is always disabled by default. Personally I think you could argue equally for either side.

I'm happy to ship with it disabled by default, but if someone has preference the other way also happy to change it.

**Comment by satya164** [comment]
LGTM

**Comment by hramos** [comment]
@facebook-github-bot shipit

**Pull request closed without merging**